Automattic/pocket-casts-android
# [Sleep Timer] Adds settings button to sleep timer bottom sheet dialog

## Description
- Adds settings button to sleep timer bottom sheet
- 26dfbbd3eb8b6e0ba5603c562b5f9d290ce8a6db I had to make a minor refactor to dynamically identify the position of the sleep timer section, so the settings open at the sleep timer section when accessed from the sleep timer bottom sheet dialog. This refactor involved creating an enum list to represent the settings items, allowing the sleep timer to be identified using this new key.
- Figma: hHFpI1RtnW4TRb448Q4Don-fi-4613_33187
- I noticed the existing sleep timer bottom sheet dialog does not match with the new specs in figma. I will address these changes in another PR


Fixes #2776

## Testing Instructions
1. Play an episode
2. Open sleep timer bottom sheet dialog
3. Tap on settings icon
4. ✅ Ensure `player_sleep_timer_settings_tapped` track event
5. ✅ Ensure General settings is opens an scrolls to Sleep timer section settings

## Screenshots or Screencast 
[Screen_recording_20240911_135901.webm](https://github.com/user-attachments/assets/6d15ebca-3fe2-497f-97d2-8e1942538b3d)

## Checklist
 
#### I have tested any UI changes...
